Native build does not work with version > 2.11.0

### Tell us the extension you're using

Client

### I tried this:

I do a native build of a quarkus cli client using v2.11.0 and quarkus 3.29.0 using
```
set -g -x GRAALVM_HOME /Users/user/git/cli/mandrel-java21-23.1.7.0-Final/Contents/Home
set -g -x JAVA_HOME /Users/user/git/cli/mandrel-java21-23.1.7.0-Final/Contents/Home
export PATH="$JAVA_HOME/bin:$PATH"
./mvnw clean package -P native
```
All works fine, but If I update to quarkus  openapi client v2.12 or 2.13 executing the resulting cli leads to
```
...
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:141)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:80)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:51)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	at java.base@21.0.7/java.lang.invoke.LambdaForm$DMH/sa346b79c.invokeStaticInit(LambdaForm$DMH)
Caused by: java.lang.RuntimeException: Error injecting io.quarkiverse.openapi.generator.providers.BaseCompositeAuthenticationProvider ch.myelexis.server.api.auth.CompositeAuthenticationProvider.compositeProvider
	at ch.myelexis.server.api.auth.CompositeAuthenticationProvider_Bean.doCreate(Unknown Source)
	at ch.myelexis.server.api.auth.CompositeAuthenticationProvider_Bean.create(Unknown Source)
	at ch.myelexis.server.api.auth.CompositeAuthenticationProvider_Bean.get(Unknown Source)
	at ch.myelexis.server.api.auth.CompositeAuthenticationProvider_Bean.get(Unknown Source)
	at io.quarkus.arc.impl.ArcContainerImpl.beanInstanceHandle(ArcContainerImpl.java:570)
	at io.quarkus.arc.impl.ArcContainerImpl.beanInstanceHandle(ArcContainerImpl.java:550)
	at io.quarkus.arc.impl.ArcContainerImpl.beanInstanceHandle(ArcContainerImpl.java:583)
	at io.quarkus.arc.impl.ArcContainerImpl.instanceHandle(ArcContainerImpl.java:545)
	at io.quarkus.arc.impl.ArcContainerImpl.instance(ArcContainerImpl.java:300)
	at io.quarkus.rest.client.reactive.runtime.RestClientBuilderImpl.register(RestClientBuilderImpl.java:313)
	at io.quarkus.rest.client.reactive.runtime.RestClientBuilderImpl.build(RestClientBuilderImpl.java:446)
	at io.quarkus.rest.client.reactive.runtime.QuarkusRestClientBuilderImpl.build(QuarkusRestClientBuilderImpl.java:279)
	at io.quarkus.rest.client.reactive.runtime.RestClientCDIDelegateBuilder.build(RestClientCDIDelegateBuilder.java:74)
	at io.quarkus.rest.client.reactive.runtime.RestClientCDIDelegateBuilder.createDelegate(RestClientCDIDelegateBuilder.java:54)
	at io.quarkus.rest.client.reactive.runtime.RestClientReactiveCDIWrapperBase.constructDelegate(RestClientReactiveCDIWrapperBase.java:75)
	at io.quarkus.rest.client.reactive.runtime.RestClientReactiveCDIWrapperBase.<init>(RestClientReactiveCDIWrapperBase.java:29)
	at ch.myelexis.server.api.ConfigApi$$CDIWrapper.<init>(Unknown Source)
	at ch.myelexis.server.api.ConfigApi$$CDIWrapper_ClientProxy.<init>(Unknown Source)
	at ch.myelexis.server.api.ConfigApi$$CDIWrapper_Bean.proxy(Unknown Source)
	at ch.myelexis.server.api.ConfigApi$$CDIWrapper_Bean.get(Unknown Source)
	at ch.myelexis.server.api.ConfigApi$$CDIWrapper_Bean.get(Unknown Source)
	... 40 more
Caused by: jakarta.enterprise.inject.CreationException: Error creating synthetic bean [5BdwMennkjygK3u19UFqxslIS-M]: java.lang.RuntimeException: Error injecting synthetic injection point of bean: qAIK5jLds34ENpbMkdxnV7nRFOc
	at io.quarkiverse.openapi.generator.providers.BaseCompositeAuthenticationProvider_5BdwMennkjygK3u19UFqxslIS-M_Synthetic_Bean.doCreate(Unknown Source)
	at io.quarkiverse.openapi.generator.providers.BaseCompositeAuthenticationProvider_5BdwMennkjygK3u19UFqxslIS-M_Synthetic_Bean.create(Unknown Source)
	at io.quarkiverse.openapi.generator.providers.BaseCompositeAuthenticationProvider_5BdwMennkjygK3u19UFqxslIS-M_Synthetic_Bean.get(Unknown Source)
	at io.quarkiverse.openapi.generator.providers.BaseCompositeAuthenticationProvider_5BdwMennkjygK3u19UFqxslIS-M_Synthetic_Bean.get(Unknown Source)
	at io.quarkus.arc.impl.CurrentInjectionPointProvider.get(CurrentInjectionPointProvider.java:48)
	... 61 more
Caused by: java.lang.RuntimeException: Error injecting synthetic injection point of bean: qAIK5jLds34ENpbMkdxnV7nRFOc
	at io.quarkiverse.openapi.generator.providers.AuthProvider_qAIK5jLds34ENpbMkdxnV7nRFOc_Synthetic_Bean.doCreate(Unknown Source)
	at io.quarkiverse.openapi.generator.providers.AuthProvider_qAIK5jLds34ENpbMkdxnV7nRFOc_Synthetic_Bean.create(Unknown Source)
	at io.quarkiverse.openapi.generator.providers.AuthProvider_qAIK5jLds34ENpbMkdxnV7nRFOc_Synthetic_Bean.get(Unknown Source)
	at io.quarkiverse.openapi.generator.providers.AuthProvider_qAIK5jLds34ENpbMkdxnV7nRFOc_Synthetic_Bean.get(Unknown Source)
	at io.quarkus.arc.impl.InstanceImpl.getBeanInstance(InstanceImpl.java:325)
	at io.quarkus.arc.impl.InstanceImpl$InstanceIterator.next(InstanceImpl.java:363)
	at java.base@21.0.7/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base@21.0.7/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1939)
	at java.base@21.0.7/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base@21.0.7/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base@21.0.7/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575)
	at java.base@21.0.7/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
	at java.base@21.0.7/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)
	at java.base@21.0.7/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622)
	at java.base@21.0.7/java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627)
	at io.quarkiverse.openapi.generator.AuthenticationRecorder.lambda$recordCompositeProvider$0(AuthenticationRecorder.java:28)
	at io.quarkiverse.openapi.generator.providers.BaseCompositeAuthenticationProvider_5BdwMennkjygK3u19UFqxslIS-M_Synthetic_Bean.createSynthetic(Unknown Source)
	... 66 more
Caused by: java.lang.NullPointerException: Unknown client
	at java.base@21.0.7/java.util.Objects.requireNonNull(Objects.java:259)
	at io.quarkus.oidc.client.runtime.AbstractTokensProducer.init(AbstractTokensProducer.java:47)
	at io.quarkiverse.openapi.generator.oidc.ReactiveOidcClientRequestFilterDelegate_Bean.doCreate(Unknown Source)
	at io.quarkiverse.openapi.generator.oidc.ReactiveOidcClientRequestFilterDelegate_Bean.create(Unknown Source)
	at io.quarkiverse.openapi.generator.oidc.ReactiveOidcClientRequestFilterDelegate_Bean.get(Unknown Source)
	at io.quarkiverse.openapi.generator.oidc.ReactiveOidcClientRequestFilterDelegate_Bean.get(Unknown Source)
	at io.quarkus.arc.impl.CurrentInjectionPointProvider.get(CurrentInjectionPointProvider.java:48)
	... 83 more
```

### This happened:

see above

### I expected this:

_No response_

### Is there a workaround?

_No response_

### How can we try to reproduce the issue?

_No response_

### Anything else?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus OpenApi version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### Community Notes

<!-- Please keep this note for the community -->
* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.name: Bug Report 🐞

--- a/client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/GeneratorProcessor.java
+++ b/client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/GeneratorProcessor.java
@@ -133,7 +133,7 @@ public class GeneratorProcessor {
     }
 
     @BuildStep
-    @Record(ExecutionTime.STATIC_INIT)
+    @Record(ExecutionTime.RUNTIME_INIT)
     void produceCompositeProviders(AuthenticationRecorder recorder, List<AuthProviderBuildItem> authProviders,
             BuildProducer<SyntheticBeanBuildItem> beanProducer) {
         Map<String, List<AuthProviderBuildItem>> providersBySpec = authProviders.stream()
@@ -149,6 +149,7 @@ public class GeneratorProcessor {
                             ParameterizedType.create(Instance.class, ClassType.create(AuthProvider.class)),
                             AnnotationInstance.builder(OpenApiSpec.class).add("openApiSpecId", openApiSpecId).build())
                     .createWith(recorder.recordCompositeProvider(openApiSpecId))
+                    .setRuntimeInit()
                     .done());
         });
     }

